wdl/tasks/Sniffles2.wdl

```diff
@@ -0,0 +1,150 @@
+version 1.0
+
+
+import "Structs.wdl"
+
+# "This workflow calls SV candidates using Sniffles2 population mode.
+# 1) Call SV candidates and create .snf file for each sample
+#  2) Combined calling using multiple .snf files into a single .vcf"
+
+workflow Sniffles2 {
+
+    input {
+        Array[File] sampleBAMs
+        Array[File] sampleBai
+        Int minsvlen
+        String prefix
+        String sample_id}
+
+    scatter (pair in zip(sampleBAMs, sampleBai)) {
+        call SampleSV {
+                input:
+                    bam = pair.left,
+                    bai = pair.right,
+                    minsvlen = minsvlen,
+                    prefix = prefix,
+                    sample_id = sample_id
+                }
+         }
+
+    call MergeCall { input: snfs = SampleSV.snf}
+
+
+    output {
+         Array[File] single_snf = SampleSV.snf
+         File multisample_vcf = MergeCall.vcf
+        }
+}
+
+
+
+task SampleSV {
+
+    input {
+        File bam
+        File bai
+        Int minsvlen
+        String? chr
+        String sample_id
+        String prefix
+        RuntimeAttr? runtime_attr_override
+    }
+
+    parameter_meta {
+        bam:              "input BAM from which to call SVs"
+        bai:              "index accompanying the BAM"
+        minsvlen:         "minimum SV length in bp. Default 35"
+        chr:              "chr on which to call variants"
+        sample_id:        "Sample ID"
+        prefix:           "prefix for output"
+    }
+
+    Int cpus = 8
+    Int disk_size = 2*ceil(size([bam, bai], "GB"))
+    String fileoutput = if defined(chr) then "~{prefix}.~{chr}.sniffles.snf" else "~{prefix}.sniffles.snf"
+    String vcf_output = if defined(chr) then "~{prefix}.~{chr}.sniffles.vcf" else "~{prefix}.sniffles.vcf"
+
+    command <<<
+        set -x
```

Comment:
we prefer to use `set -eux`

Reply:
https://gist.github.com/mohanpedala/1e2ff5661761d3abd0385e8223e16425?permalink_comment_id=3945021

Reply:
fixed